[stdlib] Rename Slab to InlineArray

Azoy opened the pull request:
In accordance with the evolution acceptance notes here: https://forums.swift.org/t/accepted-with-modifications-se-0453-inlinearray-formerly-vector-a-fixed-size-array/77678/29

Azoy committed:
Rename Slab to InlineArray

Azoy commented:
@swift-ci please test

DougGregor reviewed (approved):
Thanks!

etcwilde reviewed (approved):
Approving changes to the new build system.

Azoy committed:
Update inlinearray_literal.swift

Azoy commented:
@swift-ci please smoke test

glessard commented on lib/Sema/CSSimplify.cpp:
```diff
@@ -8709,30 +8709,30 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyConformsToConstraint(
 
-    // If we're attempting to bind an array literal to a 'Slab' parameter,
-    // then check if the counts are equal and solve.
+    // If we're attempting to bind an array literal to a n'InlineArray'
```
```suggestion
    // If we're attempting to bind an array literal to an 'InlineArray'
```

glessard commented on test/Sema/inlinearray.swift:
```diff
@@ -2,16 +2,16 @@
 
-let d: Slab<2, _> = [1, 2, 3] // expected-error {{expected '2' elements in slab literal, but got '3'}}
-let e: Slab<2, _> = [1] // expected-error {{expected '2' elements in slab literal, but got '1'}}
+let d: InlineArray<2, _> = [1, 2, 3] // expected-error {{expected '2' elements in slab literal, but got '3'}}
+let e: InlineArray<2, _> = [1] // expected-error {{expected '2' elements in slab literal, but got '1'}}
```
```suggestion
let d: InlineArray<2, _> = [1, 2, 3] // expected-error {{expected '2' elements in inline array literal, but got '3'}}
let e: InlineArray<2, _> = [1] // expected-error {{expected '2' elements in inline array literal, but got '1'}}
```

glessard replied:
I _think_ this output had been updated elsewhere?

Azoy committed:
Update lib/Sema/CSSimplify.cpp

Co-authored-by: Guillaume Lessard <glessard@tffenterprises.com>

Azoy committed:
Update test/Sema/inlinearray.swift

Co-authored-by: Guillaume Lessard <glessard@tffenterprises.com>

Azoy committed:
Update inlinearray.swift

Azoy commented:
@swift-ci please smoke test

glessard commented:
@swift-ci please smoke test

glessard commented:
@swift-ci please smoke test

glessard commented:
Looks like `Sema/typo_correction.swift`'s behaviour reverted to what it was before you changed it in 4c5a13a63e21c7b27fc9fe01ccb89e07fe95961a:
```
-  let plop = 4 // expected-note {{'plop' declared here}}
+  let plop = 4 // expected-note {{did you mean 'plop'?}}
```

glessard committed:
[test] fix typo correction

Reverses the change done to this file in 4c5a13a63e21c7b27fc9fe01ccb89e07fe95961a.

It seems like this is an unstable diagnostic.

glessard commented:
@swift-ci please smoke test
